sorc/ncep_post.fd/MDL2P.f

```diff
@@ -532,6 +533,75 @@ SUBROUTINE MDL2P(iostatusD3D)
+                 ! rain. Meanwhile, the model level below the melting
+                 ! level will contain rain. Importantly, model levels in
+                 ! this situation do not contain supercooled rain, but 
+                 ! interpolation onto the target pressure level will 
+                 ! yield supercooled rain. In this sense, the supercooled
+                 ! rain is an artifact of interpolation. Because
+                 ! supercooled rain poses a hazard to aviation, we seek 
+                 ! to prevent supercooled rain water on pressure levels 
+                 ! when none actually exists on the adjacent model levels.
+                 ! 
+                 ! In the code below, we search for the condition in
+                 ! which this artifact occurs. Then, the previously 
+                 ! interpolated value of Qrain is simply replaced by the 
+                 ! value from the overlying (subfreezing) model level, 
+                 ! where Qrain is typically zero. The same approach is 
+                 ! extended to snow and graupel, thus preserving the total
+                 ! precipitation mixing ratio locally.
+                 ! 
+                 ! Note that the code below does not eliminate all supercooled
+                 ! rain on pressure levels, nor should it. Situations in which
+                 ! supercooled rain is permitted include:
+                 !  i) The situation described by (1)-(3) above, but with
+                 !     rain water also present on the overlying (subfreezing) model 
+                 !     level. This may correspond to freezing drizzle that falls
+                 !     through a melting level (becoming ordinary drizzle). In the
+                 !     code below, the value of Qrain on the target pressure level 
+                 !     will simply be replaced by the value from the overlying 
+                 !     model level. 
+                 ! ii) Classic freezing-rain situations, characterized by a
+                 !     deep (model-resolved) surface-based refreezing layer. In
+                 !     the code below, the temperature check contained in the third 
+                 !     "IF" statement will evaluate to false.
+
+                 IF (MODELNAME == 'FV3R' .OR. MODELNAME == 'GFS') THEN
+                   IF ( (QR1(I,J) > zero) .AND. (TSL(I,J) <= TFRZ) ) THEN ! Supercooled rain water exists on this pressure level
+                     IF ( (T(I,J,LL-1) <= TFRZ) .AND. &  ! This pressure level is "near" a melting layer: temp on overlying model level is subfreezing, and
+                          (T(I,J,LL) > TFRZ) ) THEN      ! temp on underlying model level is above freezing
+                          ! Replace the interpolated mixing ratios of rain, snow, and graupel 
+                          ! on this pressure level with the values from the overlying (subfreezing) model level:
+                          QR1(I,J) = MAX(QQR(I,J,LL-1),zero)
```

Comment:
What about rain number concentration for consistency? And what about cloud water as well?

Reply:
Thanks @AndersJensen-NOAA  — It turns out that we do not include the number concentrations in our pressure-level GRIB2 output (these are offered on native levels only).  It's probably a good thing, but we could consider including them, if needed.  

I can also extend this approach to cloud water (and ice).  Are there any possible pitfalls?  Spurious supercooled rain (described by @gthompsnJCSDA ) was the original motivation, but it would seem that artificial supercooled cloud water could arise for the same reason.

Reply:
Added cloud water (and all other hydrometeors)